### PR TITLE
Get service proposed changes

### PR DESCRIPTION
**Pull request by Papkidon**
# Proposed changes

#### Structure

change overall structure of the project
rename ‘api' directory to ’src’
rename ‘\_\_tests\_\_’ directory to ’tests’
move ’tests' at the same level as ’src’ directory
rename ‘main.ts’ to ‘index.ts’
rename ‘get.ts’ to ‘router.ts'

#### Code

split responsibility of FetchMockachino class into three separate classes
split router endpoints into separate routers
move exports at the end of the files
separate devDependencies and dependencies in package.json

### Added

add status monitor at ‘metrics/status’
add unit tests
add global setup and teardown of tests

### Fixed

add correct settings in tsconfig for Node 18
change validation library from Joi to zod
change fetching library from ‘xmlhttprequest’ to built-in ‘fetch’
turn off logging while tests run
coerce ’soc’ property in ‘usage’ data to number
update Node to version 18 in Dockerfile
change npm install command to production in Dockerfile

### Removed

remove unused ‘env.logger.ts’ configuration file
remove saving logs to files

### CODE REVIEW

File: services/get-cars-service/src/config/status-monitor/status-monitor.config.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Settings for the express status monitor
+ */
+const statusMonitorConfig = {
```
Comment (DominikMarciniszyn) [inline services/get-cars-service/src/config/status-monitor/status-monitor.config.ts:4]: It is a good practice to type this monitor config

File: services/get-cars-service/src/models/cars/cars.validation.model.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod';
+import type ICars from './ICars.model';
+
+/**
+ * Validates usage data received from Mockachino API 
+ * @param data - data received from Mockachino
+ * @returns object containing information about validation
+ */
+const validateCars = (data: ICars) => {
```
Comment (DominikMarciniszyn) [inline services/get-cars-service/src/models/cars/cars.validation.model.ts:9]: There is no return type defined

File: services/get-cars-service/src/models/usage/IUsage.model.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Represents the 'usage' data schema that can be received from the Mockachino
+ * @interface IUsage
+ */
+interface IUsage {
+    vin: string,
+    carStatistics: [
+        {
+            datetime: Date,
```
Comment (DominikMarciniszyn) [inline services/get-cars-service/src/models/usage/IUsage.model.ts:9]: I prefer to exclude this element of array as a separate type. What do you think?

File: services/get-cars-service/src/models/usage/usage.validation.model.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { z } from 'zod';
+import type IUsage from './IUsage.model';
+
+/**
+ * Validates usage data received from Mockachino API 
+ * @param data - data received from Mockachino
+ * @returns object containing information about validation results
+ */
+export const validateUsage = (data: IUsage) => {
```
Comment (DominikMarciniszyn) [inline services/get-cars-service/src/models/usage/usage.validation.model.ts:9]: Missing return type

**Review by DominikMarciniszyn — CHANGES_REQUESTED** [review]
In general there is a lot of formatting changes, a few I saw, imo if you fix them it will be better clean code:
- missing empty line at the end of file
- some indentations are misleading

File: services/get-cars-service/src/models/usage/IUsage.model.ts (hunk 3 above)
Comment (Papkidon) [reply services/get-cars-service/src/models/usage/IUsage.model.ts:9]: In my opinion it depends on how big in size the nested element is and how readibility suffers when this element is not split into separate type.

File: services/get-cars-service/src/config/status-monitor/status-monitor.config.ts (hunk 1 above)
Comment (Papkidon) [reply services/get-cars-service/src/config/status-monitor/status-monitor.config.ts]: Done.

File: services/get-cars-service/src/models/cars/cars.validation.model.ts (hunk 2 above)
Comment (Papkidon) [reply services/get-cars-service/src/models/cars/cars.validation.model.ts]: Done.

File: services/get-cars-service/src/models/usage/usage.validation.model.ts (hunk 4 above)
Comment (Papkidon) [reply services/get-cars-service/src/models/usage/usage.validation.model.ts]: Done.

**Pull request merged**